### PR TITLE
Feat/pupil 210/update radio box

### PR DESCRIPTION
**Pull request by kimon-satan**
## Description

- Updated QuizMCQSingleAnswer to use the QuizRadioButton from oak-components
- Refactored QuizRender to create reusable answer type discriminators (see QuizUtils)
- Refactored handleOnChange functions in all quiz components to use customHook useInitialChange
- Created stemUtils for extracting correct stem types from arrays of stemObjects

## How to test

1. Go to https://deploy-preview-2185--oak-web-application.netlify.thenational.academy

-text only
https://deploy-preview-2185--oak-web-application.netlify.thenational.academy/pupils/programmes/combined-science-secondary-ks4-foundation-aqa/units/measuring-waves/lessons/transverse-waves
-with images
https://deploy-preview-2185--oak-web-application.netlify.thenational.academy/pupils/programmes/science-primary-ks2/units/earth-sun-and-moon/lessons/why-we-have-day-and-night#starter-quiz (second question)

## ACs

- [x]  provides feedback via OakQuizRadio
- [x]  renders single answer MCQs with images

### CODE REVIEW

**Review by benprotheroe — APPROVED** [review]

File: src/components/PupilJourneyComponents/QuizMCQSingleAnswer/QuizMCQSingleAnswer.tsx [hunk 1]
```diff
@@ -23,89 +31,44 @@ export const QuizMCQSingleAnswer = (props: QuizMCQSingleAnswerProps) => {
...
-              let feedbackModeColor: "oakGreen" | "red" | undefined;
-
-              if (questionState.feedback?.[i] === "correct") {
-                feedbackModeColor = "oakGreen";
-              } else if (questionState.feedback?.[i] === "incorrect") {
-                feedbackModeColor = "red";
-              }
+        const label = answer.answer.find((a) => a.type === "text") as
+          | StemTextObject
+          | undefined;
```
Comment (carlmw) [inline src/components/PupilJourneyComponents/QuizMCQSingleAnswer/QuizMCQSingleAnswer.tsx:51]: Looks like a job for `isText` if you export it from your new stem utils

```suggestion
        const label = answer.answer.find(isText)
```

**Review by carlmw — APPROVED** [review]
🚀

**Review by rightsaidjames — APPROVED** [review]
Looks good, although I'm pretty sure this question must be a troll?
Screenshot 2024-01-15 at 16 57 54

@kimon-satan I assume you've yet to tackle the change to feedback states? (e.g. don't tell people they were right not to pick the incorrect answers)

**Pull request merged**